Runtime/Scripts/VideoStreamTrack.cs

```diff
@@ -0,0 +1,253 @@
+        /// <param name="height"></param>
+        public VideoStreamTrack(string label, IntPtr ptr, int width, int height)
+            : base(WebRTC.Context.CreateVideoTrack(label, ptr))
+        {
+            WebRTC.Context.SetVideoEncoderParameter(self, width, height);
+            WebRTC.Context.InitializeEncoder(self);
+            tracks.Add(this);
+        }
+
+        /// <summary>
+        /// Creates from MediaStreamTrack object
+        /// </summary>
+        /// <param name="sourceTrack"></param>
+        public VideoStreamTrack(IntPtr sourceTrack) : base(sourceTrack)
+        {
+            tracks.Add(this);
+        }
+
+        public override void Dispose()
+        {
+            if (this.disposed)
+            {
+                return;
+            }
+
+            if (self != IntPtr.Zero && !WebRTC.Context.IsNull)
+            {
+                if (IsEncoderInitialized)
+                {
+                    WebRTC.Context.FinalizeEncoder(self);
+                    UnityEngine.Object.DestroyImmediate(m_destTexture);
+                }
+
+                if (m_sink != null)
+                {
+                    NativeMethods.VideoTrackRemoveSink(self, m_sink.self);
+                    m_sink.Dispose();
+                }
+
+                tracks.Remove(this);
+                WebRTC.Context.DeleteMediaStreamTrack(self);
+                self = IntPtr.Zero;
+            }
+
+            this.disposed = true;
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    public static class CameraExtension
+    {
+        public static VideoStreamTrack CaptureStreamTrack(this Camera cam, int width, int height, int bitrate,
+            RenderTextureDepth depth = RenderTextureDepth.DEPTH_24)
+        {
+            switch (depth)
+            {
+                case RenderTextureDepth.DEPTH_16:
+                case RenderTextureDepth.DEPTH_24:
+                case RenderTextureDepth.DEPTH_32:
+                    break;
+                default:
+                    throw new InvalidEnumArgumentException(nameof(depth), (int)depth, typeof(RenderTextureDepth));
+            }
+
+            int depthValue = (int)depth;
+            var format = WebRTC.GetSupportedRenderTextureFormat(SystemInfo.graphicsDeviceType);
+            var rt = new RenderTexture(width, height, depthValue, format);
+            rt.Create();
+            cam.targetTexture = rt;
+            return new VideoStreamTrack(cam.name, rt);
+        }
+
+
+        public static MediaStream CaptureStream(this Camera cam, int width, int height, int bitrate,
+            RenderTextureDepth depth = RenderTextureDepth.DEPTH_24)
+        {
+            var stream = new MediaStream(WebRTC.Context.CreateMediaStream("videostream"));
+            var track = cam.CaptureStreamTrack(width, height, bitrate, depth);
+            stream.AddTrack(track);
+            return stream;
+        }
+    }
+
+    public class UnityVideoSink : IDisposable
```

Comment:
Is this class needed define publicly?

Reply:
No, is not needed.
I change this class to internal.